apps/examples/testcase/le_tc/kernel/tc_task.c

```diff
@@ -224,6 +239,46 @@ static int vfork_task(int argc, char *argv[])
 }
 #endif
 
+/**
+* @fn                   :max_task
+* @brief                :utility function for tc_task_create
+* @return               :int
+*/
+static int max_task(int argc, char *argv[])
+{
+	int i;
+	pid_t pid = 0;
+	ppid = getpid();
+
+	/* intentionally creates more than CONFIG_MAX_TASKS */
+	for (i = 0; i < CONFIG_MAX_TASKS; i++) {
+		pid = task_create("tc_vfork_temp", SCHED_PRIORITY_MAX - 1, 1024, create_temp_task, (char * const *)NULL);
+	}
+
+	if (i == (CONFIG_MAX_TASKS)) {
```

Comment:
I don't understand a logic of checking a result. 
i.e.
We have 5 tasks and CONFIG_MAX_TASKS is 10. If this makes 4 tasks and gets ERROR as return. This TC will be passed.

How about check a number of task which is alive using sched_foreach?

Reply:
yes that can be done, but then it needs to be done under 
#ifndef CONFIG_BUILD_PROTECTED flag
This TC logic just focuses on achieving tasks greater than CONFIG_MAX_TASKS, so independent of current alive tasks, 
whenever this threshold is crossed, pid = ERROR 
after this task exits all tasks after ( i = CONFIG_MAX_TASKS ) return PID = ERROR &
all tasks created before ( i =CONFIG_MAX_TASKS) return valid PID's

Hence we dont need dependency over BUILD TYPE

Reply:
1. You didn't try to make tasks over CONFIG_MAX_TASKS. At end of TC, you didn't check how many tasks are alive. How do you say that is checking of max tasks?
2. Kishore made kernel test driver to run kernel side tc inside kernel. You should move kernel tc into kernel. Then you don't need to concern protected build.